systemtests/systemtestslib/utils.py

```diff
@@ -13,36 +14,51 @@
 CONNECTION_TIMEOUT = 600
 
 
-def get_sym_files(auth_token, url, start_page):
+def get_sym_files(baseurl, auth_token, params):
     """Given an auth token, generates filenames and sizes for SYM files.
 
-    :param auth_token: auth token for symbols.mozilla.org
-    :param url: url for file uploads
-    :param start_page: the page of files to start with
+    :param baseurl: a base url
+    :param auth_token: an auth token
+    :param params: a dict of parameters to pass by querystring.
 
     :returns: generator of (key, size) typles
 
     """
-    sym_files = []
-    page = start_page
-    params = {"page": start_page}
+    url = urljoin(baseurl.rstrip("/"), "/api/uploads/files")
     headers = {"auth-token": auth_token, "User-Agent": "tecken-systemtests"}
+    params = params or {}
+
+    sym_files = []
+    page = params.get("page", 1)
 
     while True:
-        if sym_files:
-            yield sym_files.pop(0)
-        else:
-            params["page"] = page
-            resp = requests.get(
-                url,
-                params=params,
-                headers=headers,
-                timeout=CONNECTION_TIMEOUT,
-            )
-            resp.raise_for_status()
-            data = resp.json()
-            sym_files = [(record["key"], record["size"]) for record in data["files"]]
+        params["page"] = page
+        resp = requests.get(
+            url,
+            params=params,
+            headers=headers,
+            timeout=CONNECTION_TIMEOUT,
+        )
+        resp.raise_for_status()
+        data = resp.json()
+        sym_files = [
+            (record["key"], record["size"])
+            for record in data["files"]
+            if record["completed_at"]
+        ]
+        yield from sym_files
+
+        # NOTE(willkg): sometimes a whole bunch of try uploads happen at the same
+        # time and since there's no way to filter for just regular symbols files (as
```

Comment:
Are there reasons outside of the system tests that someone would want to filter for regular symbols? Is there a bug to be able to do this?

I've seen you mention this limititation a couple of times ([other mention](https://github.com/mozilla-services/tecken/pull/2905/files#r1540332442)) in this test as being inconvenient.

Reply:
There is no bug I'm aware of. The "API" we're using here isn't an official API. It's something Peter wrote to support the React-based web frontend so the ergonomics of it are determined by the webapp ui. I haven't made any changes since I don't want to extend the API beyond what we're using in the webapp and I don't want to make changes to the webapp because we have no tests and it's antiquated. I keep hoping we redo the ui which would also cause us to redo this API.